openwisp_radius/counters/base.py

```diff
@@ -21,9 +21,17 @@ def check_name(self):  # pragma: no cover
         pass
 
     @property
-    @abstractmethod
-    def reply_name(self):  # pragma: no cover
-        pass
+    def reply_names(self):
+        # BACKWARD COMPATIBILITY: In previous versions of openwisp-radius,
+        # the Counter.reply_name was a string instead of a tuple. Thus,
+        # we need to convert it to a tuple if it's a string.
+        if not hasattr(self, "reply_name"):
```

Comment:
AFAIK, `getattr` and `hasattr` have the same computing cost, so it's better to use `getattr` if we need to use the result later.
```suggestion
        reply_name = getattr(self, "reply_name", None)
        if not reply_name:
```